src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md

```diff
@@ -0,0 +1,92 @@
+
+It was developed with an aim to use IntelliJ IDEA in most effective way. So, inspections that are running during
+the code analysis can alternatively be enabled in the [inspections settings](https://www.jetbrains.com/help/idea/code-inspection.html)
+`Preferences -> Editor -> Inspections -> UCT`:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-settings.png)
+
+It allows you to see the problematic code just in real time without running it intentionally during code writing or even files visiting.
+**The result of the real time inspection is most suitable for using during development to be always compatible with the future versions**:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-highlighting.png)
+
+The second cool IntelliJ IDEA tool under which we adapted our code by reusing **Inspections** feature of it is running inspections through special action
+`Code -> Inspect Code` or just open context menu under target directory to allow IntelliJ IDEA populates path to analyse for you automatically:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/run-inspect-code-from-context-menu.png)
+
+By calling **Inspect Code** action from the context menu we have Inspection Scope populated for us automatically.
+The last thing to configure here is **Inspection Profile**.
+By default, here will be the **Project Default** profile with all inspections that are not connected to our goal.
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-intellij-idea-based-analysis-configuration.png)
+
+So you can just create new profile with the only UCT inspections enabled (more about that [read here](https://www.jetbrains.com/help/idea/customizing-profiles.html)):
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-profile.png)
+
+For this inspection type you don't need to have UCT inspections enabled for your project.
+**The result of this inspection is most suitable for code refactoring during problematic code elimination**:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-analysis-through-intellj-idea-inspect-code-action.png)
+
+**The most powerful inspection type is the execution of compatibility inspections in a [run tool window](https://www.jetbrains.com/help/idea/run-tool-window.html) with the familiar output as in the original Adobe Commerce [Upgrade Compatibility Tool]({{site.baseurl}}/upgrade-compatibility-tool/introduction.html)**.
+
+**Advantages from using this one:**
+
+*  all problems gathered in one place
+*  there are navigations to the problem files, stored report in a json format, link to the web page with detailed description of all the error codes
+*  this report can be easily used from the merchant, management or agencies sides
+*  there is a complexity score that helps compare the upgrade complexity from practice side for agencies
```

Comment:
"practice side" - does not make sense here. Can you explain more clearly?

Reply:
Done.